src/interactivity/base/HostSignalInputThread.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "HostSignalInputThread.hpp"
+#include "../inc/HostSignals.hpp"
+
+#include "../interactivity/inc/ServiceLocator.hpp"
+
+using namespace Microsoft::Console;
+using namespace Microsoft::Console::Interactivity;
+
+// Constructor Description:
+// - Creates the PTY Signal Input Thread.
+// Arguments:
+// - hPipe - a handle to the file representing the read end of the Host Signal pipe.
+HostSignalInputThread::HostSignalInputThread(_In_ wil::unique_hfile hPipe) :

```

Comment:
you can enforce an ownership transfer requirement with `wil::unique_hfile&& hPipe`, if you so desire.